### ISSUE TITLE
Documentation CI/Action

### ISSUE THREAD

**Issue by simmsa** [MHKiT-Software/MHKiT#62]
We need better tooling for recognizing and reporting documentation build warnings and errors. #61 details the current output during `make html`

At first glance <https://github.com/marketplace/actions/sphinx-build> may be a option.

**Comment by akeeste** [comment]
@simmsa I completely agree. I think we need to:
1. Automate building of the documentation
2. Create a CI action that runs on both PRs into MHKiT-Python and MHKiT-MATLAB to test the documentation build. This would help avoid a lot of minor formatting errors as we go, and ensure that updates are always incorporated into the documentation.

**Comment by akeeste** [comment]
I can't really tell the difference between the ``Makefile`` and ``make.bat``, but when building locally on Windows my systems calls ``make.bat`` with the ``make html`` or ``make clean`` commands. This uses sphinx-build behind the scenes.

The gh-pages automation is controlled here: https://github.com/MHKiT-Software/MHKiT/settings/pages

**Comment by simmsa** [comment]
@akeeste, it seems like we have made some good progress here. Would you consider this issue fixed?

We now have working actions, working preview builds, and have a better workflow for the documentation builds in general.

**Comment by akeeste** [comment]
Agreed, this is fixed. Most of the build warnings have been resolved and we now have the better actions and previews.
